build-system/compile/compile.js

```diff
@@ -215,6 +215,11 @@ function compile(
     if (options.include3pDirectories) {
       srcs.push('3p/**/*.js', 'ads/**/*.js');
     }
+    // For ESM Builds, exclude ampdoc and ampshared css from inclusion.
+    // These styles are guaranteed to already be present on elgible documents.
+    if (options.esmPassCompilation) {
+      srcs.push('!build/ampdoc.css.js', '!build/ampshared.css.js');
+    }
```

Comment:
Sorry about the late review (I was on vacation last week). With gulp v4, I know that negative globstars need to come at the end for them to be considered, because globs are [evaluated in order](https://github.com/gulpjs/gulp/blob/master/CHANGELOG.md#vinylvinyl-fs-changes). I'm not sure if the same rule applies to negative globs that don't use a wildcard.

All negative globstars in this file appear here:

https://github.com/ampproject/amphtml/blob/3d3c31ea9987eaeab2b95d3caff49ac799c1169a/build-system/compile/compile.js#L256-L266

Something to consider, as this file grows.